op-chain-ops/interopgen/deployers/opchain.go

```diff
@@ -0,0 +1,87 @@
+package deployers
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script"
+)
+
+type DeployOPChainInput struct {
+	OpChainProxyAdminOwner common.Address
+	SystemConfigOwner      common.Address
+	Batcher                common.Address
+	UnsafeBlockSigner      common.Address
+	Proposer               common.Address
+	Challenger             common.Address
+
+	BasefeeScalar     uint32
+	BlobBaseFeeScalar uint32
+	L2ChainId         *big.Int
+	Opsm              common.Address
+}
+
+func (input *DeployOPChainInput) InputSet() bool {
+	return true
+}
+
+type DeployOPChainOutput struct {
```

Comment:
I think we have at least 3 structs that list all of the contracts now, there is the `L1Deployments` struct used by `op-e2e`, something in `superchain-ops` and this. We will need to figure out ways to have a single source of truth

Reply:
I am worried we couple the "config" and the "input" too much, where we cannot change the config (user-API) anymore without breaking existing internal API. Some duplication might separate these API layers. If that's not a concern, then I can embed these script output structs directly in the final output struct, to not duplicate their definitions.